### PR TITLE
Cleanup and documentation of Faultify.Cli

### PR DESCRIPTION
**Pull request by mbartelsm**
- Made Settings a member of the Program class, it used to be passed around everywhere.
- Made output directory into a property of Settings.
- Reworked some code for simplicity and legibility.
- Reworked an exception that was thrown at the main-level into a log-and-quit.

### CODE REVIEW

File: Faultify.Cli/Program.cs [hunk 1]
```diff
@@ -132,75 +116,83 @@ private void PrintProgress(MutationRunProgress progress)
...
         }
 
-        private async Task GenerateReport(TestProjectReportModel testResult, Settings settings)
+        /// <summary>
+        /// Builds a report based on the test results and program settings
+        /// </summary>
+        /// <param name="testResult">Report model with the test results</param>
+        private async Task GenerateReport(TestProjectReportModel testResult)
         {
-            if (string.IsNullOrEmpty(settings.ReportPath))
```
Comment (mbartelsm) [inline Faultify.Cli/Program.cs:183]: This functionality is built into the getter of ReportPath

File: Faultify.Cli/Program.cs [hunk 2]
```diff
@@ -132,75 +116,83 @@ private void PrintProgress(MutationRunProgress progress)
...
+            model.TestProjects.Add(testResult);
 
-            IReporter reporter = ReportFactory(settings.ReportType);
-            byte[] reportBytes = await reporter.CreateReportAsync(mprm);
+            IReporter reporter = ReportFactory(ProgramSettings.ReportType);
+            byte[] reportBytes = await reporter.CreateReportAsync(model);
 
             string reportFileName = DateTime.Now.ToString("yy-MM-dd-H-mm") + reporter.FileExtension;
-
-            await File.WriteAllBytesAsync(Path.Combine(_outputDirectory ?? string.Empty, reportFileName), reportBytes);
```
Comment (mbartelsm) [inline Faultify.Cli/Program.cs:196]: OutPutDirectory is never null

File: Faultify.Cli/Settings.cs [hunk 3]
```diff
@@ -16,85 +16,125 @@ namespace Faultify.Cli
     
     internal class Settings
     {
-        private static readonly Logger _logger = LogManager.GetCurrentClassLogger();
+        private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
```
Comment (mbartelsm) [inline Faultify.Cli/Settings.cs:19]: For consistency with the rest of the code

File: Faultify.Cli/Settings.cs [hunk 4]
```diff
@@ -16,85 +16,125 @@ namespace Faultify.Cli
...
+            HelpText = "The mutation level indicating the test depth.")]
         public MutationLevel MutationLevel { get; set; }
 
+        /// <summary>
+        /// The name of the test host framework. options:  "NUnit", "XUnit", "MsTest", "DotnetTest".
+        /// </summary>
         [Option('t', "testHost", Required = false, Default = nameof(TestHost.DotnetTest),
             HelpText = "The name of the test host framework.")]
-        public string TestHostName { get; set; }
+        public string _testHostName { get; set; }
```
Comment (mbartelsm) [inline Faultify.Cli/Settings.cs:61]: These fields are only public because of implementation details wrt the parser library that's being used. I think private-style naming is a good compromise to signal these shouldn't be accessed.

Comment (cdadashovkhandan) [reply Faultify.Cli/Settings.cs:61]: Honestly it feels like bad practice to me. Why not just comment that it shouldn't be accessed instead? That way the message is carried over while convention is not violated.

File: Faultify.Cli/Program.cs [hunk 5]
```diff
@@ -46,63 +62,27 @@ private static async Task Main(string[] args)
...
-                ?? throw new Exception("Couldn't get the Faultify service from the system");
-
-
-            await program.Run(settings);
-        }
-        
-        /// <summary>
-        ///     Sets up NLog configuration programmatically
-        /// </summary>
-        private static void ConfigureNLog()
```
Comment (cdadashovkhandan) [inline Faultify.Cli/Program.cs:63]: Is the functionality removed here already covered by `LogManager.Configuration = new XmlLoggingConfiguration("NLog.config");` ? If not, why was this removed?

**Review by cdadashovkhandan — COMMENTED** [review]
Good job, just need clarification on a couple of things

File: Faultify.Cli/Settings.cs (hunk 4 above)
Comment (mbartelsm) [reply Faultify.Cli/Settings.cs:61]: I am of the opposite opinion. A comment would only be visible at the declaring class, at which point it doesn't really matter because changing the value there is also possible for private variables. A name that says "I'm private" is visible anywhere.
```cs
class Settings {
    // Don't use this
    public int Var = 0

/* ... */

public void SomeMethod() {
    settings.Var = 3; // Looks good
}
``` 

```cs
class Settings {
    public int _var = 0

/* ... */

public void SomeMethod() {
    settings._var = 3; // Hmmmmm
}
```

File: Faultify.Cli/Program.cs (hunk 5 above)
Comment (mbartelsm) [reply Faultify.Cli/Program.cs:63]: Yes, it used to be hardcoded because I couldn't get the XML to work, but that's fixed now, I just forgot to remove this method back then.

File: Faultify.Cli/Settings.cs (hunk 4 above)
Comment (cdadashovkhandan) [reply Faultify.Cli/Settings.cs:61]: Good point. We can keep it then.

File: Faultify.Cli/Program.cs (hunk 5 above)
Comment (cdadashovkhandan) [reply Faultify.Cli/Program.cs:63]: Alright, nice

**Review by cdadashovkhandan — APPROVED** [review]
You can merge when ready.
